sdk/core/azure-core/azure/core/rest/_requests_basic.py

```diff
@@ -0,0 +1,132 @@
+# --------------------------------------------------------------------------
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+#
+# The MIT License (MIT)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the ""Software""), to
+# deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+# sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+from typing import TYPE_CHECKING
+
+from ..exceptions import ResponseNotReadError, StreamConsumedError, StreamClosedError
+from . import _HttpResponseBase, HttpResponse
+from ..pipeline.transport._requests_basic import StreamDownloadGenerator
+
+if TYPE_CHECKING:
+    from typing import Iterator
+
+class _RestRequestsTransportResponseBase(_HttpResponseBase):
+    def __init__(self, **kwargs):
+        super(_RestRequestsTransportResponseBase, self).__init__(**kwargs)
+        self.status_code = self.internal_response.status_code
+        self.headers = self.internal_response.headers
+        self.reason = self.internal_response.reason
+        self.content_type = self.internal_response.headers.get('content-type')
+
+    def _get_content(self):
+        """Return the internal response's content"""
+        if not self.internal_response._content_consumed:  # pylint: disable=protected-access
```

Comment:
Why if content is not consumed, we return None?

Reply:
this code was kind of confusing, so I fixed it to be less confusing. thanks!